classes/Language.php

```diff
@@ -1415,32 +1415,30 @@ public static function updateMultilangTables(Language $language, array $tablesTo
 
     public static function updateMultilangFromCldr($lang)
     {
-        $cldrLocale = $lang->getLocale();
-        $cldrFile = _PS_TRANSLATIONS_DIR_ . 'cldr/datas/main/' . $cldrLocale . '/territories.json';
-
-        if (file_exists($cldrFile)) {
-            $cldrContent = json_decode(file_get_contents($cldrFile), true);
-
-            if (!empty($cldrContent)) {
-                $translatableCountries = Db::getInstance()->executeS('SELECT c.`iso_code`, cl.* FROM `' . _DB_PREFIX_ . 'country` c
-                    INNER JOIN `' . _DB_PREFIX_ . 'country_lang` cl ON c.`id_country` = cl.`id_country`
-                    WHERE cl.`id_lang` = "' . (int) $lang->id . '" ', true, false);
-
-                if (!empty($translatableCountries)) {
-                    $cldrLanguages = $cldrContent['main'][$cldrLocale]['localeDisplayNames']['territories'];
-
-                    foreach ($translatableCountries as $country) {
-                        if (isset($cldrLanguages[$country['iso_code']]) &&
-                            !empty($cldrLanguages[$country['iso_code']])
-                        ) {
-                            $sql = 'UPDATE `' . _DB_PREFIX_ . 'country_lang`
-                                SET `name` = "' . pSQL(ucwords($cldrLanguages[$country['iso_code']])) . '"
-                                WHERE `id_country` = "' . (int) $country['id_country'] . '" AND `id_lang` = "' . (int) $lang->id . '" LIMIT 1;';
-                            Db::getInstance()->execute($sql);
-                        }
-                    }
-                }
+        $translatableCountries = Db::getInstance()->executeS('SELECT c.`iso_code`, cl.* FROM `' . _DB_PREFIX_ . 'country` c
+            INNER JOIN `' . _DB_PREFIX_ . 'country_lang` cl ON c.`id_country` = cl.`id_country`
+            WHERE cl.`id_lang` = "' . (int) $lang->id . '" ', true, false);
+
+        if (empty($translatableCountries)) {
+            return;
+        }
+
+        $container = SymfonyContainer::getInstance();
+        /** @var LocaleRepository $localeRepoCLDR */
+        $localeRepoCLDR = $container->get('prestashop.core.localization.cldr.locale_repository');
+        $localeCLDR = $localeRepoCLDR->getLocale($lang->locale);
+        $territories = $localeCLDR->getAllTerritories();
+
+        foreach ($translatableCountries as $country) {
+            if (empty($territories[$country['iso_code']])) {
+                continue;
             }
+            $territoryData = $territories[$country['iso_code']];
+            $sql = 'UPDATE `' . _DB_PREFIX_ . 'country_lang`
+                    SET `name` = "' . pSQL(ucwords($territoryData->getName())) . '"
```

Comment:
Careful with `ucword` I would advise using `mb_ucwords`

I had issues in https://github.com/PrestaShop/PrestaShop/blob/c7af233fb637eeb6ab732cd908d9c19ec4843333/src/Core/Currency/CurrencyGridDataFactory.php#L106 when using japanese because there were ideograms characters that did not like `ucfirst` very much and I ended up with little squares displayed I think it's the same with `ucwords`

Reply:
Thanks. I fixed.